lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php

```diff
@@ -205,8 +205,17 @@ private function addParametersAndGetArgs(array $freeVariables, $methodBuilder, S
             $variableType = $freeVariable->symbolContext()->type();
 
             if ($variableType->isDefined()) {
-                $parameterBuilder->type($variableType->short());
+                // TODO Create a utils class or something to get the shortest type by taking
+                // into account the import table, to handle aliases and partial namespaces
+                $shortTypeAsString = ($variableType->isNullable() ? '?' : '') . $variableType->short();
+
+                $parameterBuilder->type($shortTypeAsString);
                 if ($variableType->isClass()) {
+                    // TODO Remove this ?
+                    // We should not have to import it, we are extracting the
+                    // method into the same file. We should not try to fix an
+                    // oversight from the user, espacially since he might want
+                    // to keep his partial qualified name
                     $builder->use((string) $variableType);
```

Comment:
we need to import it -- f.e. 

- a variable was retrieved from a method
- the variable has a type
- it is now a parameter in the new method
- we need to import the type.

Can we change the comment to reflect that?

We need to fix the whole qualfiied name issue, but it's a separate thing :)

Reply:
Sorry for the delay to respond, I needed to take care of some other stuff and I don't have a lot of time left.

I'm not sure to see the case you're talking about.
Here we are extracting a method from a selection.
So the new method will always be created in the same scope: the class.
Meaning that any variable that we will have to provide as parameters to new method will already be imported in the scope.
If it is not, it's the fault of the user and we have no way of knowing which one to import in case there is multiple possibilities.

So I understand that we can try, just in case there is only one possibility.
But I don't think it's really SRP, here we should only build a method from the information at our disposal in the scope.
We should not modify the scope, even to fix an error.

If you still want to use the import I will change the comment for the one you proposed.

Reply:
> Meaning that any variable that we will have to provide as parameters to new method will already be imported in the scope.

This is not true (I previously fell for this)  -- if you call a method on an imported class, that method can return a type not in the current scope and you will therefore need to import it if it's a parameter of the extracted mehtod.